src/support/HttpClient.cpp

```diff
@@ -203,3 +208,11 @@ Vault::HttpClient::executeRequest(const Vault::Url& url,
     Vault::HttpResponseBodyString{buffer}
   });
 }
+
+void
+Vault::HttpClient::reportError(std::optional<HttpResponse> response) const {
+    nlohmann::json j;
+    j["Status code"] = response.value().statusCode.value();
+    j["Response"] = response.value().body.value();
+    responseErrorCallback_(j.dump());
+}
```

Comment:
I think this is a remnant of the original idea. I'd like to skip it and just call whatever the response error callback is when the response is not successful.

Reply:
what would you pass the responseCallback? an std::string or directly HttpResponse?

Reply:
I did the user might need both the statuscode and the body. This was the most convenient way to pack the 2 into a string. After this the user can use json parsing easily.

Reply:
I think it makes sense to use the `HttpResponse` here to allow the consumer to unpack the details as needed.